Sources/SwiftParser/Declarations.swift

```diff
@@ -1697,12 +1697,55 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+
+    let fixityText = [Keyword.prefix, .infix, .postfix].map({ $0.defaultText })
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(combining: attrs.modifiers?.elements.compactMap { $0.unexpectedBeforeName }, RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena), arena: self.arena)
+
+    var fixity: RawTokenSyntax?
+    var leftModifiers: [RawDeclModifierSyntax?] = []
+
+    if let modifiers = attrs.modifiers?.elements {
+      for modifier in modifiers {
+        if fixity == nil,
+          fixityText.contains(modifier.name.tokenText)
```

Comment:
I would prefer to use `TokenSpec` instead of comparing the token text, just because it does a bit more keyword handling and is generally considered the entry point to check if a token is of a specific kind. Something like

```swift
switch modifier.name {
case TokenSpec(.prefix), TokenSpec(.infix), TokenSpec(.postfix):
  // do something
}
```